internal/statestorage/redis/redispb/player.go

```diff
@@ -0,0 +1,177 @@
+// This can probably be deprecated if we work on getting the above generic enough.
+// The problem is that protobuf message reflection is pretty messy.
+func UnmarshalPlayerFromRedis(ctx context.Context, pool *redis.Pool, player *om_messages.Player) error {
+
+	// Get the Redis connection.
+	redisConn, err := pool.GetContext(context.Background())
+	defer redisConn.Close()
+	if err != nil {
+		pLog.WithFields(log.Fields{
+			"error":     err.Error(),
+			"component": "statestorage",
+		}).Error("failed to connect to redis")
+		return err
+	}
+
+	// Prepare redis command.
+	cmd := "HGETALL"
+	key := player.Id
+	resultLog := pLog.WithFields(log.Fields{
+		"component": "statestorage",
+		"cmd":       cmd,
+		"key":       key,
+	})
+
+	// Run redis command
+	playerMap, err := redis.StringMap(redisConn.Do(cmd, key))
+
+	// Put values from redis into the Player message
+	player.Properties = playerMap["properties"]
+	player.Pool = playerMap["pool"]
+	player.Assignment = playerMap["assignment"]
+	player.Status = playerMap["status"]
+	player.Error = playerMap["error"]
+
+	// TODO: Room for improvement here.
+	if a := playerMap["attributes"]; a != "" {
+		attrsJSON := fmt.Sprintf("{\"attributes\": %v}", a)
+		err = jsonpb.UnmarshalString(attrsJSON, player)
+		if err != nil {
+			resultLog.Error("failure on attributes")
+			resultLog.Error(a)
+		}
+	}
+
+	resultLog.Debug("state storage operation: player unmarshalled")
+	return err
+
+}
+
+// PlayerWatcher makes a channel and returns it immediately.  It also launches an
+// asynchronous goroutine that watches a redis key and returns updates to
+// that key on the channel.
+//
+// The pattern for this function is from 'Go Concurrency Patterns', it is a function
+// that wraps a closure goroutine, and returns a channel.
+// reference: https://talks.golang.org/2012/concurrency.slide#25
+//
+// NOTE: this function will never stop querying Redis during normal operation! You need to
+//  disconnect the client from the frontend API (which closes the context) once
+//  you've received the results you were waiting for to stop doing work!
+func PlayerWatcher(ctx context.Context, pool *redis.Pool, pb om_messages.Player) <-chan om_messages.Player {
+
+	pwLog := pLog.WithFields(log.Fields{"playerId": pb.Id})
+
+	// Establish channel to return results on.
+	watchChan := make(chan om_messages.Player)
+
+	go func() {
+		// var declaration
+		var prevResults = ""
+
+		// Loop, querying redis until this key has a value or the Redis query fails.
+		for {
+			select {
+			case <-ctx.Done():
+				// Player stopped asking for updates; clean up
+				close(watchChan)
+				return
+			default:
+				// Update the player's 'accessed' timestamp to denote they haven't disappeared
+				err := playerindices.Touch(ctx, pool, pb.Id)
+				if err != nil {
+					// Not fatal, but this error should be addressed.  This could
+					// cause the player to expire while still actively connected!
+					pwLog.Error("State storage error:", err.Error())
```

Comment:
Nit. "State storage error" in the playerWatcher doesn't differential well from some of the redispb errors

Reply:
Good point, updated with better message